src/gui/mzroll/tabledockwidget.h

```diff
@@ -20,320 +20,381 @@ class NumericTreeWidgetItem;
-    void clearClusters();
-
-    void writeQEInclusionList(QString fileName);
-    void writeMascotGeneric(QString fileName);
-    //void saveEICsJson(string filename);
-    //void saveEICJson(ofstream& out, EIC* eic);
-    vector<EIC *> getEICs(float rtmin, float rtmax, PeakGroup &grp);
-
-  protected:
-    void dragEnterEvent(QDragEnterEvent *event);
-    void dropEvent(QDropEvent *event);
-    void focusInEvent(QFocusEvent * event);
-    void focusOutEvent(QFocusEvent * event);
-
-  Q_SIGNALS:
-    void peakMarkedTableDock();
-    void updateProgressBar(QString, int, int);
-
-  protected Q_SLOTS:
-    void keyPressEvent(QKeyEvent *e);
-    void contextMenuEvent(QContextMenuEvent *event);
+class TableDockWidget : public QDockWidget {
+  Q_OBJECT
+
+public:
+  QWidget *dockWidgetContents;
+  QHBoxLayout *horizontalLayout;
+  QTreeWidget *treeWidget;
+  QLabel *titlePeakTable;
+  JSONReports *jsonReports;
+
+  /**
+   * @brief vallgroups will be used by libmaven/jsonReports.cpp
+   * @detail For json export. Since libmaven is written only standard
+   * cpp, all groups from <allgroups> get copied to <vallgroups> at
+   * time of json exporting
+   * @see- <TableDockWidget::exportJson>
+   */
+  vector<PeakGroup> vallgroups;
+  QList<PeakGroup> allgroups;
+
+  enum tableViewType { groupView = 0, peakView = 1 };
+  enum peakTableSelectionType { Selected = 0, Whole = 1, Good = 2, Bad = 3 };
+
+  /**
+   * @brief Construct and initialize a TableDockWidget.
+   * @detail Sets up widgets (table, toolbar with buttons, etc.) and connects
+   * any necessary signals.
+   * 
+   * @param mw A QMainWindow to which the dock belongs.
+   * @param title A string title for the dock widget.
+   * @param numColms The number of columns to be created within the table.
+   * @param type The type of groups table this is going to be.
+   */
+  TableDockWidget(MainWindow *mw);
+  ~TableDockWidget();
+
+  /**
+   * @brief Obtain the number of groups in the table.
+   * @return Integer count.
+   */
+  int groupCount() { return allgroups.size(); }
+
+  /**
+   * @brief Obtain a list of groups present in <allgroups>.
+   * @return QList of PeakGroup pointers.
+   */
+  QList<PeakGroup *> getGroups();
+
+  int tableId;
+
+  /**
+   * for making old mzroll compatible, this will act as a flag whether loaded
+   * mzroll file is old or new one. this will be set by class method
+   * <markv_0_1_5mzroll>
+   */
+  bool mzrollv_0_1_5;
+
+  /**
+   * @brief Update the name of the intensity type field in table
+   * according to the quantity type currenly selected by the user.
+   */
+  void setIntensityColName();
+
+  /**
+   * @brief Obtain the maximum intensity for a group according to
+   * the quantity type currently selected by the user.
+   * 
+   * @param group Pointer to the PeakGroup.
+   * @return Floating precision value of maximum intensity
+   */
+  float extractMaxIntensity(PeakGroup *group);
+
+  float outputRtWindow = 2.0;
```

Comment:
@saifulbkhan This is not being used anywhere.

Reply:
Even though its not being used, it could be a useful method provided by the TableDockWidget, so I let it be. I'll remove this if you think its unimportant.